src/sentry/api/serializers/models/app_platform_event.py

```diff
@@ -39,24 +39,23 @@ def get_actor(self):
 
     @property
     def body(self):
-        return {
+        return json.dumps({
             'action': self.action,
             'installation': {
                 'uuid': self.install.uuid,
             },
             'data': self.data,
             'actor': self.get_actor(),
-        }
+        })
 
     @property
     def headers(self):
-        body = json.dumps(self.body)
         request_uuid = uuid4().hex
 
         return {
             'Content-Type': 'application/json',
             'Request-ID': request_uuid,
             'Sentry-Hook-Resource': self.resource,
             'Sentry-Hook-Timestamp': six.text_type(int(time())),
-            'Sentry-Hook-Signature': self.install.sentry_app.build_signature(body)
+            'Sentry-Hook-Signature': self.install.sentry_app.build_signature(self.body)
```

Comment:
I was thinking more about this signature method and right now our checksums rely on key order, which is not going to be consistent across languages/platforms. If/when we update to python3 the implicit dictionary ordering will change and signatures will change as well.

We might want to mention in the sentry-app docs when they are created that signatures require the text based payload that we send to work. Hydrating dicts/Maps and re-serializing will only bring 😿

Reply:
Yea, is there a way to avoid that? afaik, if we're using the payload as part of the signature that's always going to be the case. We'll definitely mention in the docs that you should use the un-altered request body to verify it.